### DISCUSSION TITLE
ForecastingGridSearchCV error

### DISCUSSION THREAD

**Discussion by algorhythmist** [alan-turing-institute/sktime#2177] [Q&A] [answered]
Hi, I was searching for optimal hyperparameters using ForecastingGridSearchCV and it runs into various errors, depending on the type of data and model used, but isn't it supposed to return nan values in case of error?

I am currently using try except block but in doing so it return 0 records if a single parameter combination runs into error. If I try to set error_score as parameter I get **TypeError: __init__() got an unexpected keyword argument 'error_score'**. 

Is anyone else facing the same issue? If yes, is there a fix for it?

**Comment by aiwalter** [comment, marked as answer]
Hi @algorhythmist , pls provide a code snippet with reproducible example to reproduce your exceptions. Also please give sktime version you are using.

**Reply by algorhythmist** [reply]
Hi @aiwalter , I am currently using version 0.10.0
```python
import random
y = []
for i in range(0,50):
    n = random.randint(-30,30)
    y.append(n)

ets = {"trend" : ["multiplicative"],
       "seasonal":["multiplicative"],
       "sp":[3,6]}


cv = ExpandingWindowSplitter(initial_window =18, start_with_window=True, fh=[1,2,3])

forecaster = ExponentialSmoothing()

gscv = ForecastingGridSearchCV(forecaster=forecaster,
                                param_grid=ets,
                                cv=cv, verbose = 1, n_jobs = -1)

gscv.fit(np.array(y))
```

I have on purpose used negative values and multiplicative seasonal/trend components for Exponential Smoothing

**ValueError: endog must be strictly positive when usingmultiplicative trend or seasonal components.**

**Reply by aiwalter** [reply]
this `ValueError` comes from the model itfself and you can get rid of it by either using a `TabularToSeriesAdopter(MinMaxScaler())` in a pipeline or you change the params of  `ExponentialSmoothing`.

Pipelining: https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.forecasting.compose.TransformedTargetForecaster.html

**Reply by algorhythmist** [reply]
Thank you, but isn't the grid search expected to return nan values in case of error even from the model? I have run into various errors, for example InputArgsException while using BATS due to Box Cox transforms

**Reply by aiwalter** [reply]
no, we dont catch the internal model errors separately and I also dont understand why it should return NaN. If it would return NaN, you would not know where the error was coming from and what it was about to fix it.

**Reply by algorhythmist** [reply]
I was referring to this from the documentation about returning nan for errors

**error_score: numeric value or the str ‘raise’, optional (default=np.nan)
The test score returned when a forecaster fails to be fitted.**

**Reply by aiwalter** [reply]
which doc?

**Reply by algorhythmist** [reply]
https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html

**Reply by aiwalter** [reply]
ah I see, there are 2 params that are not implemented but given in docstring:
```
    error_score: numeric value or the str 'raise', optional (default=np.nan)
        The test score returned when a forecaster fails to be fitted.
    return_train_score: bool, optional (default=False)
```
Could you kindly create a BUG report where you desribe this, we should probably implement sth here.

**Reply by algorhythmist** [reply]
@aiwalter sure, I will create a bug report, thank you for all your help, I was checking source code of earlier versions and this was implemented there

**Reply by aiwalter** [reply]
which version?

**Reply by algorhythmist** [reply]
I only checked v0.4.2

https://www.sktime.org/en/v0.4.2/modules/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html?highlight=gridsearch#sktime.forecasting.model_selection.ForecastingGridSearchCV

**Reply by d-vasek** [reply]
> no, we dont catch the internal model errors separately and I also dont understand why it should return NaN. If it would return NaN, you would not know where the error was coming from and what it was about to fix it.

Hello, just wanted to add some more info to this. Having the possibility to switch between NaN and raising error would be highly appreciated. The reason for that is that if I run grid search with many models and params and just one of these many combinations fails after hours of training, I have no result available and all the computation time is wasted. Having the possibility to have NaNs in the results list allows to inspect problematic model-param combinations without losing the results from all other valid model trainings. The possibility to raise an error could still be kept there, if somebody would prefer that behavior.

**Reply by fkiraly** [reply]
> I only checked v0.4.2

we are currently on v0.11.0, that's 7 non-patch since then
